src/coreComponents/linearAlgebra/utilities/LAIHelperFunctions.hpp

```diff
@@ -157,6 +159,127 @@ void SeparateComponentFilter( MATRIX const & src,
   dst.close();
 }
 
+template< typename VECTOR >
+void ComputeRigidBodyModes( MeshLevel const & mesh,
+                            DofManager const & dofManager,
+                            std::vector< std::string > const & selection,
+                            array1d< VECTOR > & rigidBodyModes )
+{
+  NodeManager const & nodeManager = *mesh.getNodeManager();
+
+  localIndex numComponents = 0;
+  array1d< globalIndex > globalNodeList;
+  for( localIndex k = 0; k < LvArray::integerConversion< localIndex >( selection.size() ); ++k )
+  {
+    if( dofManager.getLocation( selection[k] ) == DofManager::Location::Node )
+    {
+      string const & dispDofKey = dofManager.getKey( selection[k] );
+      arrayView1d< globalIndex const > const & dofNumber = nodeManager.getReference< globalIndex_array >( dispDofKey );
+      localIndex const numComponentsField = dofManager.numComponents( selection[k] );
+      numComponents = numComponents > 0 ? numComponents : numComponentsField;
+      GEOSX_ERROR_IF( numComponents != numComponentsField, "Rigid body modes called with different number of components." );
+      globalIndex const globalOffset = dofManager.globalOffset( selection[k] );
```

Comment:
```suggestion
      globalIndex const globalOffset = dofManager.globalOffset( selection[k] );
      globalIndex const numLocalDofs = LvArray::integerConversion< globalIndex >( dofManager.numLocalDofs( selection[k] ) );
```

Reply:
I don't understand this suggestion ... sorry!